File: src/IsbInterop/ServiceForms/IServiceFactory.vb
Namespace ServiceForms

  ''' <summary>
  ''' Фабрика служебных объектов.
  ''' </summary>
  Public Interface IServiceFactory
    Inherits IIsbComObjectWrapper

    ''' <summary>
    ''' Получает проводник системы.
    ''' </summary>
    ''' <param name="isMain">Признак получения главной формы проводника системы.</param>
    ''' <returns>Проводник системы.</returns>
    Function GetExplorer(isMain As Boolean) As IEdmsExplorer
  End Interface
End NameSpace

File: src/IsbInterop/ServiceForms/IServiceFactory.cs
namespace IsbInterop.ServiceForms
{
  /// <summary>
  /// Фабрика служебных объектов.
  /// </summary>
  public interface IServiceFactory : IBaseIsbObject
  {
    /// <summary>
    /// Получает проводник системы.
    /// </summary>
    /// <param name="isMain">Признак получения главной формы проводника системы.</param>
    /// <returns>Проводник системы.</returns>
    IEdmsExplorer GetExplorer(bool isMain);
  }
}
